.github/workflows/benchmarks.yml

```diff
@@ -0,0 +1,69 @@
+name: Benchmarks
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+        - '**'
+  pull_request:
+    branches:
+      - '**'
+
+concurrency:
+  # SHA is added to the end if on `main` to let all main workflows run
+  group: ${{ github.ref }}-${{ github.workflow }}-${{ github.event_name }}-${{ github.ref == 'refs/heads/main' && github.sha || '' }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: Benchmarks
+    runs-on: benchmark
+    timeout-minutes: 30
+    strategy:
+      fail-fast: false
+      max-parallel: 4
+      matrix:
+        python-version: [3.7, 3.8, 3.9]
```

Comment:
maybe this is unnecessary. we wouldn't really expect the python version to matter much

Reply:
Historically we've been bitten on the move to the next version. I lean that we drop to just 3.9 and then add 3.10 once we're ready and keep a pattern of newest and one behind.